workspaces/topology/plugins/topology/src/components/Topology/permissions/MissingPermissionPage.tsx

```diff
@@ -57,42 +59,50 @@ type MissingPermissionPageProps = { permissions: Permission[] };
 export const MissingPermissionPage = ({
   permissions,
 }: MissingPermissionPageProps) => {
+  const { t } = useTranslation();
   const navigate = useNavigate();
 
+  const permissionNames = permissions.map(perm => perm.name).join(', ');
+
+  const permissionText =
+    permissions.length === 1
+      ? t('permissions.permission')
+      : t('permissions.permissions');
+
   return (
     <StyledBox>
       <Grid container alignItems="center" justifyContent="space-around">
         <Grid item xs={4} md={4}>
           <Stack spacing={3} alignItems="flex-start">
             <Typography variant="h3" style={{ fontWeight: 400 }}>
-              Missing Permission
+              {t('permissions.missingPermission')}
             </Typography>
             <Typography variant="body1">
-              To view Topology, your administrator must grant you{' '}
-              {permissions.map((perm, i) => {
-                const isLast = i === permissions.length - 1;
-                const isSecondLast = i === permissions.length - 2;
-                let separator = ', ';
-                if (isSecondLast) separator = ' and ';
-                if (isLast) separator = ' ';
+              {(() => {
+                const TOKEN = '__PERMS__';
+                const sentence = t('permissions.missingPermissionDescription', {
+                  permissions: TOKEN,
+                  permissionText,
```

Comment:
Building sentences like this might not work in 'all' languages. We can keep that for now but let us track this as a bug so that we can improve this in the next round.

One option could be to use Markdown instead.

```tsx
const permissionNames = "**" + permissions.map(perm => perm.name).join('**, **') + "**";
```

And then we should check if we can use https://www.i18next.com/translation-function/plurals for one permission vs many permissions.